Plugins/Mute/DataManager.cs

```diff
@@ -1,38 +1,64 @@
-﻿using Data.Models.Client;
+﻿using SharedLibraryCore.Database.Models;
 using SharedLibraryCore.Interfaces;
 using static System.Enum;
 
 namespace Mute;
 
 public class DataManager
 {
+    private readonly IMetaServiceV2 _metaService;
+
     public DataManager(IMetaServiceV2 metaService)
     {
         _metaService = metaService;
     }
 
-    private readonly IMetaServiceV2 _metaService;
-
-    public async Task<MuteState> ReadPersistentData(EFClient client)
+    public async Task<MuteStateMeta> ReadPersistentData(EFClient client)
     {
-        var clientMuteState = client.GetAdditionalProperty<MuteState?>(Plugin.MuteKey) ??
-                              Parse<MuteState>((await _metaService.GetPersistentMeta(Plugin.MuteKey, client.ClientId))?
-                                  .Value ?? nameof(MuteState.Unmuted));
+        var clientMuteStateMeta = client.GetAdditionalProperty<MuteStateMeta>(Plugin.MuteKey) ??
+                                  await _metaService.GetPersistentMetaValue<MuteStateMeta>(Plugin.MuteKey,
+                                      client.ClientId);
+        
+        if (clientMuteStateMeta is not null) return clientMuteStateMeta;
+        if (clientMuteStateMeta is null)
+        {
+            var tryParse = TryParse<MuteState>(
+                (await _metaService.GetPersistentMeta(Plugin.MuteKey, client.ClientId))?.Value,
+                out var clientMuteState);
+
 
-        client.SetAdditionalProperty(Plugin.MuteKey, clientMuteState);
-        return clientMuteState;
+            clientMuteStateMeta = new MuteStateMeta
+            {
+                ClientId = client.ClientId,
+                Reason = string.Empty,
+                Expiration = null,
+                AdminId = 0,
+                MuteState = tryParse ? clientMuteState : MuteState.Unmuted
+            };
+        }
+
+        client.SetAdditionalProperty(Plugin.MuteKey, clientMuteStateMeta);
```

Comment:
I'm not a fan of this architecturally. The read method should not create a new instance. 
And honestly you should not have a meta state until you've either muted or unmuted someone.
There is nothing preventing you from performing the correct logic if the mute state is null. 
eg on connect if they don't have a mute state (null) you don't need to do anything.